File: src/algorithms/arrays/validate-subsequence/solution2.py
# O(n) time | O(1) space
def isValidSubsequence(array, sequence):
    seqIdx = 0
    for value in array:
    	if seqIdx == len(sequence):
    		break
    	if sequence[seqIdx] == value:
    		seqIdx += 1
    return seqIdx == len(sequence)
    

File: src/algorithms/arrays/validate-subsequence/solution2.go
package main

// O(n) time | O(1) space
func IsValidSubsequence(array []int, sequence []int) bool {
	seqIdx := 0
	for _, value := range array {
		if seqIdx == len(sequence) {
			break
		}
		if value == sequence[seqIdx] {
			seqIdx += 1
		}
	}
	return seqIdx == len(sequence)
}